checks.d/docker_daemon.py

```diff
@@ -0,0 +1,640 @@
+
+        Requires _filter_containers to run first.
+        """
+        return container.get('_is_filtered', False)
+
+    def _report_container_size(self, containers_by_id):
+        container_list_with_size = None
+        for container in containers_by_id.itervalues():
+            if self._is_container_excluded(container):
+                continue
+
+            tags = self._get_tags(container, PERFORMANCE)
+
+            if "SizeRw" in container:
+                self.gauge('docker.container.size_rw', container['SizeRw'],
+                    tags=tags)
+            if "SizeRootFs" in container:
+                self.gauge(
+                    'docker.container.size_rootfs', container['SizeRootFs'],
+                    tags=tags)
+
+    def _report_image_size(self, images):
+        for image in images:
+            tags = self._get_tags(image, IMAGE)
+            if 'VirtualSize' in image:
+                self.gauge('docker.image.virtual_size', image['VirtualSize'], tags=tags)
+            if 'Size' in image:
+                self.gauge('docker.image.size', image['Size'], tags=tags)
+
+    # Performance metrics
+
+    def _report_performance_metrics(self, containers_by_id):
+        for container in containers_by_id.itervalues():
+            if self._is_container_excluded(container) or not self._is_container_running(container):
+                continue
+
+            tags = self._get_tags(container, PERFORMANCE)
+            self._report_cgroup_metrics(container, tags)
+            self._report_net_metrics(container, tags)
+
+    def _report_cgroup_metrics(self, container, tags):
+        try:
+            for cgroup in CGROUP_METRICS:
+                stat_file = self._get_cgroup_file(cgroup["cgroup"], container['Id'], cgroup['file'])
+                stats = self._parse_cgroup_file(stat_file)
+                if stats:
+                    for key, (dd_key, metric_func) in cgroup['metrics'].iteritems():
+                        if key in stats:
+                            metric_func(self, dd_key, int(stats[key]), tags=tags)
+
+                    # Computed metrics
+                    for mname, (key_list, fct, metric_func) in cgroup.get('to_compute', {}).iteritems():
+                        values = [stats[key] for key in key_list if key in stats]
+                        if len(values) != len(key_list):
+                            self.log.debug("Couldn't compute {0}, some keys were missing.".format(mname))
+                            continue
+                        value = fct(*values)
+                        if value is not None:
+                            metric_func(self, mname, value, tags=tags)
+
+
+
+        except MountException as ex:
+            if self.cgroup_listing_retries > 3:
+                raise ex
+            else:
+                self.warning("Couldn't find the cgroup files. Skipping the CGROUP_METRICS for now."
+                             "Will retry a few times before failing.")
```

Comment:
We should replace "a few" with the actual number of additional retries we're going to do. No reason to be vague when we know the exact value